packages/presentation/presentation_publishing/interfaces/fetch/publishes_unified_search.ts

```diff
@@ -113,15 +121,27 @@ export function useSearchApi({
   }, []);
 
   useEffect(() => {
-    searchApi.filters$.next(filters);
+    if (
+      !onlyDisabledFiltersChanged(searchApi.filters$.getValue(), filters, {
+        ...COMPARE_ALL_OPTIONS,
+        // do not compare $state to avoid refreshing when filter is pinned/unpinned (which does not impact results)
+        state: false,
+      })
+    ) {
+      searchApi.filters$.next(filters);
+    }
   }, [filters, searchApi.filters$]);
 
   useEffect(() => {
-    searchApi.query$.next(query);
+    if (!fastIsEqual([searchApi.query$.getValue()], [query])) {
+      searchApi.query$.next(query);
+    }
   }, [query, searchApi.query$]);
 
   useEffect(() => {
-    searchApi.timeRange$.next(timeRange);
+    if (!fastIsEqual([searchApi.timeRange$.getValue()], [timeRange])) {
```

Comment:
Quick question - why are you putting the objects into arrays for the comparison rather than just using `fastIsEqual` on the objects themselves?

Reply:
I've always seen the array format, but I've changed with [a67aa08](https://github.com/elastic/kibana/pull/203150/commits/a67aa08950213089d090445508b89f5cc260b099) your suggestion